File: Frontend/UserTech/src/app/dto/reserved-bike-dto.ts
import {StationDTO} from './station-dto';

export interface ReservedBikeDTO {
  id: string;
  station: StationDTO;
  reservedAt: string;
  reservedTill: string;
}

export interface ReservedBikesDTO {
  bikes: ReservedBikeDTO[];
}

File: Libraries/ClassLibrary/DTO/ReservedBikeDTO.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace ClassLibrary.DTO
{
    public class ReservedBikeDTO
    {
        public string BikeID { get; set; }
        public StationDTO Station { get; set; }

        public DateTime ReservedAt { get; set; }
        public DateTime ReservedTill { get; set; }
    }
}
